frontend/lib/src/components/widgets/TextArea/TextArea.tsx

```diff
@@ -150,53 +146,22 @@ const TextArea: FC<Props> = ({
     setFocused(true)
   }, [])
 
-  const onChange = useCallback(
-    (e: React.ChangeEvent<HTMLTextAreaElement>): void => {
-      const { value } = e.target
-      const { maxChars } = element
-
-      if (maxChars !== 0 && value.length > maxChars) {
-        return
-      }
-
-      // mark it dirty but don't update its value in the WidgetMgr
-      // This means that individual keypresses won't trigger a script re-run.
-      setUiValue(value)
-      setDirty(true)
-    },
-    [element]
+  const onChange = useOnInputChange(
+    element.formId,
+    element.maxChars,
+    setDirty,
+    setUiValue,
+    setValueWithSource
   )
 
-  const isEnterKeyPressed = (
-    event: React.KeyboardEvent<HTMLTextAreaElement>
-  ): boolean => {
-    const { keyCode, key } = event
-
-    // Using keyCode as well due to some different behaviors on Windows
-    // https://bugs.chromium.org/p/chromium/issues/detail?id=79407
-    return (
-      (key === "Enter" || keyCode === 13 || keyCode === 10) &&
-      // Do not send the sentence being composed when Enter is typed into the IME.
-      !(event.nativeEvent?.isComposing === true)
-    )
-  }
-
-  const onKeyDown = useCallback(
-    (e: React.KeyboardEvent<HTMLTextAreaElement>): void => {
-      const { metaKey, ctrlKey } = e
-      const { formId } = element
-      const allowFormEnterToSubmit = widgetMgr.allowFormEnterToSubmit(formId)
-
-      if (isEnterKeyPressed(e) && (ctrlKey || metaKey) && dirty) {
-        e.preventDefault()
-
-        commitWidgetValue({ fromUi: true })
-        if (allowFormEnterToSubmit) {
-          widgetMgr.submitForm(formId, fragmentId)
-        }
-      }
-    },
-    [element, widgetMgr, dirty, commitWidgetValue, fragmentId]
+  console.log("widgetMgr", widgetMgr)
```

Comment:
Can this be removed?

Reply:
oopsies, thank you! I have also created a tech debt item to enable eslint to catch unintended `console.log` statements automatically